osu.Game/Rulesets/UI/IHitObjectContainer.cs

```diff
@@ -19,6 +19,6 @@ public interface IHitObjectContainer
         /// <remarks>
         /// If this <see cref="IHitObjectContainer"/> uses pooled objects, this is equivalent to <see cref="Objects"/>.
         /// </remarks>
-        IEnumerable<DrawableHitObject> AliveObjects { get; }
+        IList<DrawableHitObject> AliveObjects { get; }
```

Comment:
I'm aware that this should be an `IReadonlyList` for safety. Unfortunately that interface doesn't implement `IndexOf`, for reasons that are beyond me. Open to suggestions here.

Reply:
I don't like this change one bit and think it should be reverted (along with the change in `HitObjectContainer`), Now calling `AliveObjects` is a silent list copy.

If `CheckHittable()` in osu! needs `IndexOf()`, it should dump `AliveObjects.ToList()` locally.

Reply:
it was stupid of me not to do this in the first place honestly, now that you've pointed it out. have changed.